pytorch_lightning/utilities/cli.py

```diff
@@ -76,6 +83,64 @@ def add_lightning_class_args(
             instantiate=not issubclass(lightning_class, Trainer),
         )
 
+    def add_optimizer_args(
+        self,
+        optimizer_class: Union[Type[Optimizer], Tuple[Type[Optimizer], ...]],
+        nested_key: str = 'optimizer',
+        link_to: str = 'AUTOMATIC',
+    ) -> None:
+        """
+        Adds arguments from an Optimizer class to a nested key of the parser
+
+        Args:
+            optimizer_class: Any subclass of torch.optim.Optimizer.
+            nested_key: Name of the nested namespace to store arguments.
+            link_to: Dot notation of a parser key to set arguments or AUTOMATIC.
+        """
+        if isinstance(optimizer_class, tuple):
+            assert all(issubclass(o, Optimizer) for o in optimizer_class)
+        else:
+            assert issubclass(optimizer_class, Optimizer)
+        kwargs = {
+            'instantiate': False,
+            'fail_untyped': False,
+            'skip': {'params'},
+        }
+        if isinstance(optimizer_class, tuple):
+            self.add_subclass_arguments(optimizer_class, nested_key, required=True, **kwargs)
+        else:
+            self.add_class_arguments(optimizer_class, nested_key, **kwargs)
+        self.optimizers_and_lr_schedulers[nested_key] = (optimizer_class, link_to)
+
+    def add_lr_scheduler_args(
+        self,
+        lr_scheduler_class: Union[Type[_LRScheduler], Tuple[Type[_LRScheduler], ...]],
+        nested_key: str = 'lr_scheduler',
+        link_to: str = 'AUTOMATIC',
+    ) -> None:
+        """
+        Adds arguments from an _LRScheduler class to a nested key of the parser
+
+        Args:
+            optimizer_class: Any subclass of torch.optim.lr_scheduler._LRScheduler.
+            nested_key: Name of the nested namespace to store arguments.
+            link_to: Dot notation of a parser key to set arguments or AUTOMATIC.
+        """
+        if isinstance(lr_scheduler_class, tuple):
+            assert all(issubclass(o, _LRScheduler) for o in lr_scheduler_class)
```

Comment:
I think this is not True for all LRSChedulers (like ReduceOnPlateau)

Reply:
Oh, that is unfortunate. `ReduceOnPlateau` only inherits from object. The simple fix is to remove all reliance on `_LRScheduler`. Or is there some other idea to keep some minimal check for the given scheduler classes?

Reply:
Yeah that's a really bad case. For now I think we could check if it is a subclass of `_LRScheduler` or `ReduceOnPlateau`. However, we may have to update this in the future if someone requests it.

Reply:
We already impose that restriction here

https://github.com/PyTorchLightning/pytorch-lightning/blob/55a90af7fc0805855684e93dbad669f5bbe76eee/pytorch_lightning/trainer/optimizers.py#L160-L172

Reply:
Fixed